src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health, biten) {
+    super(name, health);
+
+    this.biten = biten;
+  }
+
+  bite(object) {
+    if (object.hidden === false) {
```

Comment:
```suggestion
    if (!object.hidden) {
```

also, you should check `instanceof Herbivore`